packages/layer-composer/src/generators/vessel-events/vessel-events.ts

```diff
@@ -68,39 +51,66 @@ class VesselsEventsGenerator {
       })
     }
 
-    const source: VesselsEventsSource = {
-      id,
+    const pointsSource: VesselsEventsSource = {
+      id: `${id}_points`,
       type: 'geojson',
       data: newData,
     }
-    return [source]
+
+    if (!config.showTrackSegments) {
```

Comment:
```suggestion
    if (!config.zoom <= POINTS_TO_SEGMENTS_ZOOM_LEVEL_SWITCH) {
```
This is one of the potential issues with performance. Segments are calculated even though zoom allows only for displaying points.